scripts/update-cdn-urls.js

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env node
+/**
+ * @fileoverview
+ * Updates the CDN URLs in the README.md to the latest major version. Does
+ * nothing if the README.md is already up-to-date.
+ *
+ * It assumes the major version in the README is either equal to the current
+ * package major version, or the package minor version minus 1. If not the
+ * behaviour of this script should be consider undefined.
+ */
+
+const fs = require("fs");
+const path = require("path");
+const getMajorVersion = require("semver/functions/major");
+
+const rootDir = path.resolve(__dirname, "..");
+const packageJsonFile = path.resolve(rootDir, "package.json");
+const readmeFile = path.resolve(rootDir, "README.md");
+
+function getManifest() {
+  const manifestRaw = fs.readFileSync(packageJsonFile).toString();
+  const manifestJson = JSON.parse(manifestRaw);
+  return manifestJson;
```

Comment:
Why not just return in the `JSON.parse` line? The function `getManifest` should be the readable logic wrapper itself.

Reply:
No strong opinions on this one. I will wait with changing it until we decided on your other comment.

My reasoning for doing it this way: I have found that it makes future diffs easier to read as the `return` statement doesn't have to change or move if the logic of the function needs to change. You may think of it this way: the function must return a `manifestJson` variable, and I don't really care so much how it was obtained. That said, I'm not consistently using this style and this function is not a particular good example/use case of this principle.


> The function getManifest should be the readable logic wrapper itself.

I don't see how the extra line makes it less readable, unless you count longer as harder.

Reply:
> You may think of it this way: the function must return a `manifestJson` variable, and I don't really care so much how it was obtained.

Should not be named `getManifestJson` the function if that is always supposed to return that? Apart from that, this would really change in the future? It is just to read the `package.json` file, I don't think we are going to change the way to get the version.

Reply:
> Should not be named getManifestJson the function if that is always supposed to return that?

That is a good point, but at a different scope. My reasoning concerns only the function as it's scope. Don't get me wrong, I don't have any data or "official" guidelines to back me up, it's just something I've been playing with lately.


> Apart from that, this would really change in the future? It is just to read the `package.json` file, I don't think we are going to change the way to get the version.

Definitely agree, but you never know 😉 Even in my very short journey as a software engineer I've seen it happen many times that this was said and turned out not to be true.